engine/src/test/java/io/camunda/zeebe/engine/processing/incident/BusinessRuleTaskIncidentTest.java

```diff
@@ -65,6 +65,36 @@ private IncidentRecordValueAssert assertIncidentCreated(
   // --------------------------------------------------------------------------
   // ----- CalledDecision related tests
 
+  @Test
+  public void shouldCreateIncidentIfDecisionNotDeployed() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            processWithBusinessRuleTask(
+                b -> b.zeebeCalledDecisionId("unknown_decision_id").zeebeResultVariable("result")))
+        .deploy();
+
+    // when
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    final var taskActivating =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATING)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementId(TASK_ELEMENT_ID)
+            .withElementType(BpmnElementType.BUSINESS_RULE_TASK)
+            .getFirst();
+
+    // then
+    assertIncidentCreated(processInstanceKey, taskActivating.getKey())
+        .hasErrorType(ErrorType.CALLED_DECISION_ERROR)
+        .hasErrorMessage(
+            """
+            Expected to evaluate decision 'unknown_decision_id', \
+            but no decision found for id 'unknown_decision_id'\
+            """);
```

Comment:
:x: In addition, we should verify that we can resolve this incident by deploying the missing DMN.

Since we distinguish both incident error types, we should verify for both that we can resolve them.
To make it independent from the deployment of other tests, we could scope the `EngineRule` per test case instead of per test class.